### DISCUSSION TITLE
How to parse `cdr` sent from ROS2(Humble) foxglove_bridge in FoxgloveClient

### DISCUSSION THREAD

**Discussion by byebaibai** [foxglove/community#642] [Q&A] [answered]
I have a customed MSG sent from the server in ROS2(Humble) by foxglove_bridge, 
```
std_msgs/Header header
uint8 secret
uint64 birth_timestamp
uint64 timestamp
float64[8] data
string[8] names
```
and I tried to parse it in the` FoxgloveClient` in TypeScript, but failed with `cdr` encoding type.

May I ask any suggestions about parsing `cdr` encoded data?

**Comment by jtbandes** [comment, marked as answer]
You can use https://github.com/foxglove/rosmsg2-serialization.

**Reply by byebaibai** [reply]
Thanks so much!
